publishing/datasetAPI/put_version_test.go

```diff
@@ -436,15 +440,75 @@ func TestFailureToUpdateVersion(t *testing.T) {
 		}
 	})
 
+	// test for updating a version so it with a state of `published` but missing mandatory fields
+	Convey("Given a published dataset and edition and an unpublished version exist ...", t, func() {
+
+		Convey("... with mandatory fields missing", func() {
+			docs, err := setupResources(datasetID, editionID, edition, instanceID, 9)
+			if err != nil {
+				log.ErrorC("Was unable to setup test data", err, nil)
+				os.Exit(1)
+			}
+
+			Convey("When an authorised PUT request is made to update version resource to published", func() {
+				Convey("Then fail to update resource and return a status bad request (400) with a message `Missing collection_id for association between version and a collection`", func() {
```

Comment:
it doesnt feel like the end of this message matches what is being asserted in Contains() - this could just say 'with the correct message' or it should test for both aspects of the outcome if theres genuinely two messages (same below)